pkg/serving/config_changes.go

```diff
@@ -269,3 +344,200 @@ func removeEnvVars(env []corev1.EnvVar, toRemove []string) []corev1.EnvVar {
+				insertConfigMapSet[envSrc.ConfigMapRef.Name] = true
+			}
+		}
+
+		if envSrc.SecretRef != nil {
+			if _, ok := insertSecretSet[envSrc.SecretRef.Name]; ok {
+				insertSecretSet[envSrc.SecretRef.Name] = true
+			}
+		}
+	}
+
+	for k, v := range insertConfigMapSet {
+		if !v {
+			envFromSources = append(envFromSources, corev1.EnvFromSource{
+				ConfigMapRef: &corev1.ConfigMapEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: k,
+					}}})
+		}
+	}
+
+	for k, v := range insertSecretSet {
+		if !v {
+			envFromSources = append(envFromSources, corev1.EnvFromSource{
+				SecretRef: &corev1.SecretEnvSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: k,
+					}}})
+		}
+	}
+
+	return envFromSources, nil
+}
+
+func removeEnvFrom(envFromSources []corev1.EnvFromSource, toRemove []string) ([]corev1.EnvFromSource, error) {
+	for _, name := range toRemove {
+		volumeSourceType, volumeSourceName, err := parseVolumeSource(name)
+		if err != nil {
+			return nil, err
+		}
+		for i, envSrc := range envFromSources {
+			if (volumeSourceType == ConfigMapVolumeSourceType && envSrc.ConfigMapRef != nil && volumeSourceName == envSrc.ConfigMapRef.Name) ||
+				(volumeSourceType == SecretVolumeSourceType && envSrc.SecretRef != nil && volumeSourceName == envSrc.SecretRef.Name) {
+				envFromSources = append(envFromSources[:i], envFromSources[i+1:]...)
+				break
+			}
+		}
+	}
+	return envFromSources, nil
+}
+
+type volumeMountInfo struct {
+	volumeSourceType VolumeSourceType
+	volumeSourceName string
+	mountPath        string
+}
+
+func parseVolumeMountsString(s string) (*volumeMountInfo, error) {
+	slices := strings.SplitN(s, "@", 2)
+	if len(slices) != 2 {
+		return nil, fmt.Errorf("Argument requires a value that contains the @ character; got %q", s)
+	}
+
+	volumeSourceType, volumeSourceName, err := parseVolumeSource(slices[0])
+	if err != nil {
+		return nil, err
+	}
+	mountPath := slices[1]
+	return &volumeMountInfo{
+		volumeSourceType: volumeSourceType,
+		volumeSourceName: volumeSourceName,
+		mountPath:        mountPath,
+	}, nil
+}
+
+func updateVolume(volume *corev1.Volume, info *volumeMountInfo) error {
+	switch info.volumeSourceType {
+	case ConfigMapVolumeSourceType:
+		volume.Secret = nil
+		volume.ConfigMap = &corev1.ConfigMapVolumeSource{LocalObjectReference: corev1.LocalObjectReference{Name: info.volumeSourceName}}
+	case SecretVolumeSourceType:
+		volume.ConfigMap = nil
+		volume.Secret = &corev1.SecretVolumeSource{SecretName: info.volumeSourceName}
+	default:
+		return fmt.Errorf("Invalid VolumeSourceType")
+	}
+	return nil
+}
+
+func updateVolumeMountsFromMap(volumes []corev1.Volume, volumeMounts []corev1.VolumeMount, mounts map[string]string) ([]corev1.Volume, []corev1.VolumeMount, error) {
+	updatedInVolumes := make(map[string]bool)
+	updatedInVolumeMounts := make(map[string]bool)
+	parsedMap := make(map[string]*volumeMountInfo)
```

Comment:
`parsedMap` might be better named `mountInfoByName`

Reply:
I agree!

Reply:
While splitting the flag `--volume-mount` into `--volume` and `--volume-mount`, this part is disappeared.